orc/src/main/java/org/apache/iceberg/orc/ORCSchemaUtil.java

```diff
@@ -0,0 +1,412 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.orc;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.TypeUtil;
+import org.apache.iceberg.types.Types;
+import org.apache.orc.TypeDescription;
+
+/**
+ * Utilities for mapping Iceberg to ORC schemas.
+ */
+final class ORCSchemaUtil {
+
+  private static final String ICEBERG_ID_ATTRIBUTE = "iceberg.id";
+  private static final String ICEBERG_REQUIRED_ATTRIBUTE = "iceberg.required";
+
+  private static final Map<Type.TypeID, TypeDescription.Category> TYPE_MAPPING =
+      ImmutableMap.<Type.TypeID, TypeDescription.Category>builder()
+          .put(Type.TypeID.BOOLEAN, TypeDescription.Category.BOOLEAN)
+          .put(Type.TypeID.INTEGER, TypeDescription.Category.INT)
+          .put(Type.TypeID.TIME, TypeDescription.Category.INT)
+          .put(Type.TypeID.LONG, TypeDescription.Category.LONG)
+          .put(Type.TypeID.FLOAT, TypeDescription.Category.FLOAT)
+          .put(Type.TypeID.DOUBLE, TypeDescription.Category.DOUBLE)
+          .put(Type.TypeID.DATE, TypeDescription.Category.DATE)
+          .put(Type.TypeID.TIMESTAMP, TypeDescription.Category.TIMESTAMP)
+          .put(Type.TypeID.STRING, TypeDescription.Category.STRING)
+          .put(Type.TypeID.UUID, TypeDescription.Category.BINARY)
+          .put(Type.TypeID.FIXED, TypeDescription.Category.BINARY)
+          .put(Type.TypeID.BINARY, TypeDescription.Category.BINARY)
+          .put(Type.TypeID.DECIMAL, TypeDescription.Category.DECIMAL)
+          .build();
+
+  private ORCSchemaUtil() {}
+
+  static TypeDescription toOrc(Schema schema) {
```

Comment:
The convention in other util classes is to call this `convert` because it converts from one representation to the other. That would distinguish this from the method that builds a projection of an ORC file schema.